packages/app-page-builder/src/editor/components/Editor/Content.tsx

```diff
@@ -81,24 +77,17 @@ const BaseContainer = styled(LegacyBaseContainer)`
     container-name: page-editor-canvas;
 `;
 
-const Content = () => {
-    const rootElementId = useRecoilValue(rootElementAtom);
-    const rootElement = useRecoilValue(elementsAtom(rootElementId)) as PbEditorElement;
-    const [{ displayMode }, setUiAtomValue] = useRecoilState(uiAtom);
+const Content = makeDecoratable("Content", () => {
+    useDebugUtilities();
+    const rootElement = useRootElement();
+    const { displayMode, setPagePreviewDimensions } = useDisplayMode();
     const pagePreviewRef = useRef<HTMLDivElement>(null);
 
-    const setPagePreviewDimension = useCallback(
-        pagePreviewDimension => {
-            setUiAtomValue(prev => setPagePreviewDimensionMutation(prev, pagePreviewDimension));
-        },
-        [uiAtom]
-    );
-
     const resizeObserver = useMemo(() => {
         return new ResizeObserver((entries: ResizeObserverEntry[]) => {
             for (const entry of entries) {
                 const { width, height } = entry.contentRect;
-                setPagePreviewDimension({ width, height });
+                setPagePreviewDimensions({ width, height });
```

Comment:
Why do we even need this resizeObserver? I know I've been removing some of these in the past, when switching to new rendering engine.

But maybe this one is still required. Can you remind me?

Reply:
Looks like this one can be removed. There's still one for BC in the `useResponsiveClassName` hook, which is only there for legacy rendering engine (so it seems, because we don't use that hook any more).